### ISSUE TITLE
Improvement for k8s.io/docs/tutorials/kubernetes-basics/create-cluster/

### ISSUE THREAD

**Issue by sjmudd** [kubernetes/website#24355]
I guess there are various options out there. I've been using microk8s which seems to work well. Perhaps you can provide some options of different alternatives to using minikube?

The other thing which is not 100% clear is that for people who want to manage their own k8s for self-tuition or other reasons some descriptions of the different options, their limitations and so on would be useful.  One thing I've struggled with is storage, that is to do something even privately you need local storage and the different options seem to be quite varied and the landscape also changing quite quickly. However, to use k8s privately for adhoc personal use if you need storage you need something as no app is useful if no state can **safely** be stored.

So some better help or description of some of the options would be good.

**Comment by sftim** [comment]
I suggest revising https://k8s.io/docs/tutorials/kubernetes-basics/create-cluster so that it hyperlinks to https://kubernetes.io/docs/setup/learning-environment/. How does that sound @sjmudd?

If you also want https://k8s.io/docs/setup/learning-environment/ expanded to list the different options of those two in-project learning environments, that's feasible - please log a separate issue.

*Note* that if you want a comparison that also includes 3rd party software such as Microk8s, that sort of page is kind of against the [content guide](https://kubernetes.io/docs/contribute/style/content-guide/). The reason for not taking on that kind of comparison is that the project would get a _lot_ of PRs from vendors keen to get listed, and the review capacity would not be there to match demand. It'd be lovely to have a team of unbiased reviewers ready to impose a neutral point of view, and with the time to make that happen, but that's not the reality today.

/language en

**Comment by sftim** [comment]
If anyone's keen to write up these options, especially from a clearly neutral perspective, the *blog team* would love to review your PR.

(Blog articles are always a point-in-time item so it's OK if they aren't maintained once written. We also have a presumption against revising the meaning and conclusions for blog articles.)

**Comment by sjmudd** [comment]
So to respond to earlier comments I understand the desire to stay neutral and not have to list lost of different offerings.
However, my primary concern is not with getting minikube or microk8s working (at. least basically) it's about how to create it with a storage option which is more permanent. That is specifically about the storage administration.  If you do anything on k8s that requires persistent, safe storage then I believe minikube does not really solve that problem. The descriptions of PV and PVC objects is good but personally I'd like to associate, reasonably transparently a storage class with something that allows me to store data more permanently. Real focus: database storage and moving databases not in containerland into k8s.

If you use the cloud provider storage options this is pretty well handled but that seems less so if you want to use your own storage. Again my focus here is for testing / home usage but even here when "playing" you don't want to lose your data.

I know this is still a hot topic and changing quite quickly but most of the "getting started" documentation I see is focussed on apps that do not have storage, or there are options which give simple storage options (setup a single MySQL container, setup wordpress) and in these examples the details of more complex usage and storage administration is hardly touched on.

Perhaps what I'm asking for is not appropriate, but I have looked around and not found what I need. I have bumped into things like trying openebs lead me to have certificate problems which blocked me, again due to inexperience with k8s.  I had thought about using NFS or iSCSI but haven't found an obvious way to auto-provision storage (volume) on demand from kubernetes as needed using the simpler storage classes.  It may be there, but I have not understood how to set it up.

So thanks for listening. k8s is a complex piece of software and you guys have done a lot of work to make it easy to use.  Thanks.  The storage side of this seems like one of the harder pieces and while the user side is reasonably well handled with PV / PVCs the administration side seems to assume you are comfortable with the rest of the environment.

**Comment by sftim** [comment]
I'm not sure that the topic you have in mind is a good fit for the section https://k8s.io/docs/tutorials/kubernetes-basics/

“more complex” is right. That said, I'm sure the website would benefit from a guide to learning how to move on. Pages that explain how to configure, for example, _n_ local VMs with storage shared between them, or 5 credit-card-sized computers that talk iSCSI and do things with block devices.

@sjmudd can you summarize what you'd like in a phrase short enough to be a new title for this issue? I'd like to capture the essence of what you're asking and then actually amend that title.

**Comment by sjmudd** [comment]
I guess what I'm looking for is "How to help people move on from basics to more complex tasks/topics" where my specific focus is "storage", but I can see other topics like "security" or "networking" or "moving from a single server to multiple (=HA)" and so on.  On one's own it's quite hard to absorb all of this in one go, even if the high level abstract topics are understandable. What's needed are enough of the lower level details in a few typical scenarios.

Does that answer your question?

**Comment by sftim** [comment]
> As a Kubernetes learner
> I want to understand how to deploy a training environment that includes advanced features
>   …such as custom cluster networking, replicated storage, or enhanced security controls
> So that I can continue my learning

How about that?
